sycl/doc/extensions/experimental/sycl_ext_intel_fp_control.asciidoc

```diff
@@ -0,0 +1,171 @@
+to features and gather community feedback. Interfaces defined in this specification
+are implemented in {dpcpp}, but they are not finalized and may change incompatibly in
+future versions of {dpcpp} without prior notice. **Shipping software products should not
+rely on APIs defined in this specification.**
+
+== Backend support status
+
+This extension is currently implemented in {dpcpp} only for Intel GPU devices and
+only when using the Level Zero backend or OpenCL backend. Also currently Intel GPU
+driver supports this extension only for Vector Backend, so feature is supported only
+for ESIMD.
+Attempting to use this extension in kernels that run on other devices or
+backends may result in undefined behavior.  Be aware that the compiler
+is not able to issue a diagnostic to warn you if this happens.
+
+== Overview
+
+Intel GPUs allow to control kernel behavior in floating point operations. This extension adds the kernel property `fp_control` which provides a way to specify the rounding mode, denorm mode and single precision floating point mode for floating point operations in a SYCL kernel.
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_INTEL_FP_CONTROL` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== Properties
+
+|===
+|Property|Description
+
+|`fp_control`
+|The `fp_control` kernel property provides a way to specify the rounding mode, denorm mode and single precision floating point mode for a kernel.
+|===
+
+The values that the property can be set to are a subset of the following that can be OR'ed together:
+
+[%header,cols="1,5"]
+|===
+|Mode | Description
+| round_to_nearest | Round to nearest or even
+| round_upward | Round towards positive infinity
+| round_downward | Round towards negative infinity
+| round_toward_zero | Round towards zero
+| denorm_ftz | Denorm mode flush to zero
+| denorm_d_allow | Denorm mode double allow
+| denorm_f_allow | Denorm mode float allow
+| denorm_hf_allow | Denorm mode half allow
+| denorm_allow | Denorm mode double/float/half allow
+|===
+
+The compiler will not enforce using mutually exclusive values (for instance 2 rounding
+modes) and the results in this case may be undefined. At most one `fp_control` property may be associated with a kernel. If a kernel is not associated with a `fp_control` property, the manner in which the fp mode is selected is implementation-defined.
```

Comment:
This seems easy for the compiler to enforce by using `static_assert` in the header.  Can we add this error checking?

Reply:
Enforced this by using static_assert and updated doc.